src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -719,22 +719,41 @@ public static bool AreReferenceAssignable(Type dest, Type src)
         public static bool IsSameOrSubclass(Type type, Type subType) =>
             AreEquivalent(type, subType) || subType.IsSubclassOf(type);
 
-        public static void ValidateType(Type type, string paramName) => ValidateType(type, paramName, -1);
+        public static void ValidateType(Type type, string paramName) => ValidateType(type, false, false, paramName);
 
-        public static void ValidateType(Type type, string paramName, int index)
+        public static void ValidateType(Type type, bool allowByRef, bool allowPointer, string paramName)
```

Comment:
The "paramName" argument should stay in the number 2 argument position (as with the other overload) with the boolean arguments coming after.

This will also make it easier to write self documenting calls (ValidateType(type, paramName, allowByRef: true, allowPointer:true);